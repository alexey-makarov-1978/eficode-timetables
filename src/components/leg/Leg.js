import React from "react";
import PropTypes from "prop-types";

import { getTime, getDistance } from "../utils/utils";

import Grid from "@material-ui/core/Grid";
import { makeStyles } from "@material-ui/core/styles";
import TripOriginIcon from "@material-ui/icons/TripOrigin";
import MoreVertIcon from "@material-ui/icons/MoreVert";

const Leg = ({ data }) => {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <Grid container>
        <Grid item xs={2}>
          <b>{getTime(data.startTime)}</b>
        </Grid>
        <Grid item xs={2}>
          <TripOriginIcon />
        </Grid>
        <Grid item xs={8}>
          <b>{data.from.name}</b>
        </Grid>
        <Grid item xs={2}></Grid>
        <Grid item xs={2}>
          <MoreVertIcon />
        </Grid>
        <Grid item xs={8}>
          {data.mode +
            (data.route ? " (" + data.route.shortName + ")" : "") +
            " " +
            getDistance(data.distance) +
            " (" +
            Math.floor(data.duration / 60) +
            " min)"}
        </Grid>
        <Grid item xs={2}>
          <b>{getTime(data.endTime)}</b>
        </Grid>
        <Grid item xs={2}>
          <TripOriginIcon />
        </Grid>
        <Grid item xs={8}>
          <b>{data.to.name}</b>
        </Grid>
      </Grid>
    </div>
  );
};

const useStyles = makeStyles({
  root: {
    backgroundColor: "gainsboro",
    flexGrow: 3,
    padding: 2,
  },
});

Leg.propTypes = {
  data: PropTypes.object.isRequired,
};

export default Leg;
