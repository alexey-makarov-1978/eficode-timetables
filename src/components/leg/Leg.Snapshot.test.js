import React from "react";
import Leg from "./Leg";
import renderer from "react-test-renderer";

it("Leg matching snapshot", () => {
  const leg = {
    startTime: 1598869261000,
    endTime: 1598869715000,
    mode: "TRAM",
    duration: 454.0,
    distance: 1733.578241151397,
    route: {
      shortName: "1",
    },
    from: {
      name: "Hanken",
    },
    to: {
      name: "Kaupunginpuutarha",
    },
  };

  const tree = renderer.create(<Leg data={leg} />);
  expect(tree).toMatchSnapshot();
});
