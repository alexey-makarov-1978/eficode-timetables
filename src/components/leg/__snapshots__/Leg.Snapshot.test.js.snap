// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leg matching snapshot 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="MuiGrid-root MuiGrid-container"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
    >
      <b>
        13:21
      </b>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M2 12C2 6.48 6.48 2 12 2s10 4.48 10 10-4.48 10-10 10S2 17.52 2 12zm10 6c3.31 0 6-2.69 6-6s-2.69-6-6-6-6 2.69-6 6 2.69 6 6 6z"
        />
      </svg>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
    >
      <b>
        Hanken
      </b>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
    />
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
        />
      </svg>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
    >
      TRAM (1) 1 km (7 min)
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
    >
      <b>
        13:28
      </b>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M2 12C2 6.48 6.48 2 12 2s10 4.48 10 10-4.48 10-10 10S2 17.52 2 12zm10 6c3.31 0 6-2.69 6-6s-2.69-6-6-6-6 2.69-6 6 2.69 6 6 6z"
        />
      </svg>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
    >
      <b>
        Kaupunginpuutarha
      </b>
    </div>
  </div>
</div>
`;
